pinot-perf/src/main/java/org/apache/pinot/perf/BenchmarkNativeVsLuceneTextIndex.java

```diff
@@ -0,0 +1,267 @@
+        .addSingleValueDimension(INT_COL, FieldSpec.DataType.INT)
+        .build();
+    SegmentGeneratorConfig config = new SegmentGeneratorConfig(tableConfig, schema);
+    config.setOutDir(INDEX_DIR.getPath());
+    config.setTableName(TABLE_NAME);
+    config.setSegmentName(SEGMENT_NAME_NATIVE);
+    config.setFSTIndexType(fstType);
+
+    SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
+    try (RecordReader recordReader = new GenericRowRecordReader(rows)) {
+      driver.init(config, recordReader);
+      driver.build();
+    }
+  }
+
+  private ImmutableSegment loadLuceneSegment()
+      throws Exception {
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
+    Set<String> textIndexCols = new HashSet<>();
+    textIndexCols.add(DOMAIN_NAMES_COL_LUCENE);
+    indexLoadingConfig.setTextIndexColumns(textIndexCols);
+    Set<String> invertedIndexCols = new HashSet<>();
+    invertedIndexCols.add(DOMAIN_NAMES_COL_LUCENE);
+    indexLoadingConfig.setInvertedIndexColumns(invertedIndexCols);
+    return ImmutableSegmentLoader.load(new File(INDEX_DIR, SEGMENT_NAME_LUCENE), indexLoadingConfig);
+  }
+
+  private ImmutableSegment loadNativeIndexSegment()
+      throws Exception {
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
+    Map<String, String> propertiesMap = new HashMap<>();
+    FSTType fstType = FSTType.NATIVE;
+    propertiesMap.put(FieldConfig.TEXT_FST_TYPE, FieldConfig.TEXT_NATIVE_FST_LITERAL);
+
+    Map<String, Map<String, String>> columnPropertiesParentMap = new HashMap<>();
+    Set<String> textIndexCols = new HashSet<>();
+    textIndexCols.add(DOMAIN_NAMES_COL_NATIVE);
+    indexLoadingConfig.setTextIndexColumns(textIndexCols);
+    indexLoadingConfig.setFSTIndexType(fstType);
+    Set<String> invertedIndexCols = new HashSet<>();
+    invertedIndexCols.add(DOMAIN_NAMES_COL_NATIVE);
+    indexLoadingConfig.setInvertedIndexColumns(invertedIndexCols);
+    columnPropertiesParentMap.put(DOMAIN_NAMES_COL_NATIVE, propertiesMap);
+    indexLoadingConfig.setColumnProperties(columnPropertiesParentMap);
+    return ImmutableSegmentLoader.load(new File(INDEX_DIR, SEGMENT_NAME_NATIVE), indexLoadingConfig);
+  }
+
+  @Benchmark
+  @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+  public void query(Blackhole bh) {
+    if (_fstType.equalsIgnoreCase("lucene")) {
+      _indexSegment = _luceneSegment;
+      _queryContext = _luceneQueryContext;
+    } else if (_fstType.equalsIgnoreCase("native")) {
+      _indexSegment = _nativeIndexSegment;
+      _queryContext = _nativeQueryContext;
+    } else {
+      throw new IllegalStateException("Unknown value seen");
+    }
+
+    Operator<?> operator = _planMaker.makeSegmentPlanNode(_indexSegment, _queryContext).run();
+    bh.consume(operator);
+    for (int i = 0; i < _numBlocks; i++) {
+      operator.nextBlock();
+      bh.consume(operator.nextBlock());
```

Comment:
this is suspicious. either blackhole them both or just call next block once per iteration.